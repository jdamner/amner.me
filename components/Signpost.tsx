"use client";

import React from "react";
/* API */
import { useState } from "react";
import {
  motion,
  AnimatePresence,
  useReducedMotion,
  type HTMLMotionProps,
} from "framer-motion";

/* Components */
import Button from "./Atoms/Links/Button";

export default function Signpost({
  title,
  children,
}: {
  children?: React.ReactNode;
  title: string;
}) {
  const [isOpen, setIsOpen] = useState(false);
  const reducedMotion = useReducedMotion();

  const handleModalClick = () => {
    setIsOpen(!isOpen);
  };

  const motionProps: HTMLMotionProps<"div"> = reducedMotion
    ? {}
    : {
        initial: { y: 100, opacity: 0 },
        animate: { y: 0, opacity: 1 },
        exit: { y: 100, opacity: 0 },
        transition: { duration: 0.15, type: "spring" },
      };

  return (
    <>
      <Button onClick={handleModalClick} data-testid="contact-button">
        {title}
      </Button>
      <AnimatePresence>
        {isOpen && (
          <div
            className="fixed z-10"
            aria-labelledby="modal-title"
            role="dialog"
            aria-modal="true"
          >
            <div className="fixed inset-0 bg-gray-900 bg-opacity-75 transition-opacity">
              <div className="flex min-h-full items-end justify-center p-4 text-center sm:items-center sm:p-0">
                <motion.div
                  {...motionProps}
                  className="fixed overflow-hidden sm:my-8 sm:w-full sm:max-w-lg bg-white dark:bg-slate-800 text-slate-800 dark:text-slate-100"
                >
                  {/* Close Icon */}
                  <div className="absolute top-0 right-0 pt-4 pr-4">
                    <button
                      type="button"
                      className="active:outline-none focus:outline-none"
                      onClick={handleModalClick}
                    >
                      <span className="sr-only">Close</span>
                      <svg
                        className="w-6 h-6"
                        xmlns="http://www.w3.org/2000/svg"
                        fill="none"
                        viewBox="0 0 24 24"
                        stroke="currentColor"
                        aria-hidden="true"
                      >
                        <path
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeWidth="2"
                          d="M6 18L18 6M6 6l12 12"
                        />
                      </svg>
                    </button>
                  </div>
                  {/* Content */}
                  {children}
                </motion.div>
              </div>
            </div>
          </div>
        )}
      </AnimatePresence>
    </>
  );
}
