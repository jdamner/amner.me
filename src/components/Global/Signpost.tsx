/* API */
import { useState } from "react";
import { event } from "../../api/Insights";

/* Components */
import Button from "../Links/Button";


/* Types */
import type { children } from "../../types/children.type";

/**
 * Signpost Modal
 * 
 * @param { string, children } props
 * @returns {JSX.Element}
 */
export default function Signpost({ title, children }: { title: string, children: children}): JSX.Element {
    const [isOpen, setIsOpen] = useState(false);

    const handleModalClick = () => {
        setIsOpen(!isOpen);
        event('signpost-click', { open: !isOpen });
    }

    return (
        <>
            <Button onClick={handleModalClick} data-testid='contact-button'>{title}</Button>
            {isOpen &&
                <div className="fixed z-10" aria-labelledby="modal-title" role="dialog" aria-modal="true">
                    <div className="fixed inset-0 bg-gray-900 bg-opacity-75 transition-opacity"></div>
                    <div className="fixed inset-0 z-10 overflow-y-auto">
                        <div className="flex min-h-full items-end justify-center p-4 text-center sm:items-center sm:p-0">
                            <div className="relative transform overflow-hidden sm:my-8 sm:w-full sm:max-w-lg bg-white dark:bg-slate-800 text-slate-800 dark:text-slate-100">
                                {/* Close Icon */}
                                <div className="absolute top-0 right-0 pt-4 pr-4">
                                    <button type="button" className="active:outline-none focus:outline-none" onClick={handleModalClick}>
                                        <span className="sr-only">Close</span>
                                        <svg className="w-6 h-6" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true">
                                            <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M6 18L18 6M6 6l12 12" />
                                        </svg>
                                    </button>
                                </div>
                                {/* Content */}
                                {children}
                            </div>                            
                        </div>
                    </div>
                </div>
            }
        </>
    )
}