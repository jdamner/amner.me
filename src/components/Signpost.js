import React, { Component, Fragment } from "react";
import ReactModal from "react-modal";

import Corner from '../assets/corner1.svg';

export default class Signpost extends Component {
    
    constructor(props) {
        super(props);
        this.state = {
            isOpen: false,
        }
    }

    modalStyle = {
        content: {
            top: '50%',
            left: '50%',
            right: 'auto',
            bottom: 'auto',
            marginRight: '-50%',
            transform: 'translate(-50%, -50%)',
            border: '4px solid black',
            zIndex: '200',
          },
        overlay: {
            backgroundColor: 'rgba(0, 0, 0, 0.75)',
        }
    }

    handleModalClick = () => {
        this.setState({
            isOpen: !this.state.isOpen
        });
    }

    handleModalClose = () => {
        this.setState({
            isOpen: false
        });
    }

    render = () => {
        ReactModal.setAppElement('#___gatsby')
        return (
            <Fragment>
                <button className='signpost' onClick={this.handleModalClick} >
                    <div className='signpost-top'></div>
                    <div className='signpost-content'>
                        <span>Contact</span>
                    </div>
                    <div className='signpost-bottom'></div>
                </button>
                <ReactModal
                    isOpen={this.state.isOpen}
                    onRequestClose={this.handleModalClose}
                    contentLabel="Contact"
                    style={this.modalStyle}
                >
                    <section className='contact' id='contact'>
                        <Corner />
                        <div className='contact-title'>
                            <h2>Say Hello</h2>
                            <p>Looking for someone just like me?</p>
                        </div>
                        <div className='contact-list'>
                            <a className='btn' href='tel:+447515352631'>
                                +44 07515 352631
                            </a>
                            <a className='btn' href='mailto:jdamner@me.com'>
                                jdamner@me.com
                            </a>
                        </div>
                    </section>
                </ReactModal>
            </Fragment>
        )
    }
}