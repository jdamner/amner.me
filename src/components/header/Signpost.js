import React, { Component, Fragment } from "react";
import { Link } from "gatsby";
import ReactModal from "react-modal";

import Corner from '../../assets/corner1.svg';

import Style from "./Style";

export default class Signpost extends Component {

    constructor(props) {
        super(props);
        this.state = {
            isOpen: false,
        }
    }

    handleModalClick = () => {
        this.setState({
            isOpen: !this.state.isOpen
        });
    }

    handleModalClose = () => {
        this.setState({
            isOpen: false
        });
    }

    render = () => {
        return (
            <Fragment>
                <button className='signpost' onClick={this.handleModalClick} >
                    <div className='signpost-top'></div>
                    <div className='signpost-content'>
                        <span>Contact</span>
                    </div>
                    <div className='signpost-bottom'></div>
                </button>
                <ReactModal
                    appElement={document.getElementById('___gatsby')}
                    isOpen={this.state.isOpen}
                    onRequestClose={this.handleModalClose}
                    contentLabel="Contact"
                    style={Style}
                >
                    <section className='contact' id='contact'>
                        <Corner />
                        <div className='contact-title'>
                            <h2>Say Hello</h2>
                            <p>Looking for someone just like me?</p>
                        </div>
                        <div className='contact-list'>
                            <a href='tel:+447515352631'>
                                +44 07515 352631
                            </a>
                            <a href='mailto:jdamner@me.com' className=''>
                                jdamner@me.com
                            </a>
                        </div>
                    </section>
                </ReactModal>
            </Fragment>
        )
    }
}