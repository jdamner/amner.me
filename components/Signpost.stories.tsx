import type { Meta, StoryObj } from "@storybook/react";

import Signpost from "./Signpost";

type Story = StoryObj<typeof Signpost>;

export default {
  title: "Components/Signpost",
  component: Signpost,
  argTypes: {
    children: {
      control: {
        type: "none",
      },
    },
  },
} as Meta;

export const Default: Story = {
  args: {
    title: "Signpost",
    children: (
      <div
        style={{
          display: "flex",
          flexDirection: "column",
          justifyContent: "center",
          gap: "1rem",
          minHeight: "200px",
        }}
      >
        Content
      </div>
    ),
  },
};
