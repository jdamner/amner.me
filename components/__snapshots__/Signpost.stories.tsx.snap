// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Signpost Default smoke-test 1`] = `
<div class="bg-slate-100 dark:bg-slate-900 text-slate-900 dark:text-slate-300">
  <button data-testid="contact-button"
          class="undefined border-2 border-slate-500 uppercase font-bold bg-white dark:bg-slate-900 px-4 py-2 text-slate-800 dark:text-slate-200 hover:bg-slate-100 dark:hover:bg-slate-800 hover:text-black dark:hover:text-white"
  >
    Signpost
  </button>
</div>
`;
